### PR TITLE
Require that spec.subscriber is set for Subscription.

### PR DESCRIPTION
**Pull request by evankanderson**
Followup from [this comment](https://github.com/knative/specs/pull/25#issuecomment-885147111); we're going to require `spec.subscriber` on both Triggers and Channels; `spec.reply` is optional for Channels.

### CODE REVIEW

File: specs/eventing/control-plane.md [hunk 1]
```diff
@@ -230,11 +230,9 @@ condition MUST indicate the reason for the error. The Subscription MUST also set
...
 `spec` reference cannot be resolved.
 
-At least one of `spec.subscriber` and `spec.reply` MUST be set; if only
-`spec.reply` is set, the behavior is equivalent to setting `spec.subscriber`
-except that the Channel SHOULD NOT
-[advertise the ability to process replies](data-plane.md#derived-reply-events)
-during the delivery.
+The `spec.subscriber` destination MUST be set; if the `spec.reply` field is not
+set, [replies](data-plane.md#derived-reply-events) from the `spec.subscriber`
+are discarded.
```
Comment (duglin) [inline specs/eventing/control-plane.md:235]: MUST be discarded

**Comment by duglin** [comment]
One nit, otherwise looks good

**Comment by n3wscott** [comment]
This changes channels, and we might need to make a migration script or bump the version of the Subscription CRD to support this without bricking clusters.

**Comment by duglin** [comment]
approved on the TMC Sept 2 call.
/LGTM
/approve

**Pull request merged**